warp/src/raygun/mod.rs

```diff
@@ -334,127 +334,262 @@ impl Ord for MessagePage {
     }
 }
 
-#[derive(Debug, Hash, Clone, Copy, Serialize, Deserialize, PartialEq, Eq, Display)]
-#[serde(rename_all = "lowercase")]
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq, Hash, Display)]
+#[serde(rename_all = "snake_case")]
```

Comment:
Should this be untagged if its flatten internally?

Reply:
> Should this be untagged if its flatten internally?

That's up to the user if it's flattened or not. warps-ipfs is flattening it but we shouldn't assume that here.

Reply:
Oh, I see what you mean. hmm.. this will break the serialized format for sure. Let me think about it..

Reply:
I actually can't think of a good solution here since [untagged repr](https://serde.rs/enum-representations.html#untagged) means serde will have to do a trial-and-error when deserializing and it can get it wrong. We can use the internally-tagged rep though but that's still breaking the serialized form. 

Give that:
1. we have added fields in these structs and serialized formats recently.
2. this PR removes a field (`conversation_type`).

Do we really need to worry about the serialized format changing? 🤔

Reply:
Could probably get by with using `#[serde(tag = "type")]` 

> Do we really need to worry about the serialized format changing?

Since this PR is using `conversations` in `ConversationDocument`, we would want to think about the formatting too since not all `serde` implementation will have the same representation and could easily error when serializing/deserializing or break (part of the reason why things are decoupled from what is shown in `warp` vs `warp-ipfs`, granted there are still some items that need to be resolved like `Message` being used internally, which #437 will also resolve).

Reply:
> Could probably get by with using `#[serde(tag = "type")]`

That's the internally-tagged repr and as I wrote, that will still break the serialized format. 

> could easily error when serializing/deserializing

How so? I don't see the connection between failures in (de)serialization here. `ConversationDocument` even shares the `Conversation` field types, e.g `ConversationSettings` and `ConversationType` currently. If there was a connection, it would equally apply to these duplicated fields.

> or break

As I wrote:

> 1. we have added fields in these structs and serialized formats recently.

i-e we do break the serialized format whenever we need to, it seems.